tests/testsuite/doc.rs

```diff
@@ -1543,3 +1543,72 @@ fn crate_versions() {
 
     assert!(doc_html.contains("Version 1.2.4"));
 }
+
+#[cargo_test]
+fn crate_versions_flag_is_overridden_by_rustdocflags() {
+    // Testing unstable flag
+    if !is_nightly() {
+        return;
+    }
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "foo"
+            version = "1.2.4"
+            authors = []
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("-Z crate-versions doc")
+        .masquerade_as_nightly_cargo()
+        .env("RUSTDOCFLAGS", "-Z unstable-options --crate-version 2.0.3")
+        .run();
+
+    let doc_file = p.root().join("target/doc/foo/index.html");
+    let mut doc_html = String::new();
+    File::open(&doc_file)
+        .unwrap()
+        .read_to_string(&mut doc_html)
+        .unwrap();
+
+    assert!(!doc_html.contains("1.2.4"));
+    assert!(doc_html.contains("Version 2.0.3"));
+}
+
+#[cargo_test]
+fn crate_versions_flag_is_overridden_by_rustdoc_extra_arguments() {
```

Comment:
I don't think it is necessary to define a whole separate test.  To avoid the repetition, at the bottom of the other test, call `p.build_dir().rm_rf();` and then `p.cargo("rustdoc …")`, and then check the results again (can maybe put the asserts in a closure so they aren't repeated).